pxtsim/runtime.ts

```diff
@@ -38,12 +38,13 @@ namespace pxsim {
         }
 
         export function perfNow(): number {
-            const perf = performance.now               ||
-                       (performance as any).mozNow     ||
-                       (performance as any).msNow      ||
-                       (performance as any).oNow       ||
-                       (performance as any).webkitNow  ||
-                       Date.now;
+            const perf = typeof performance != "undefined" ?
+                        performance.now ||
+                        (performance as any).mozNow    ||
+                        (performance as any).msNow     ||
+                        (performance as any).oNow      ||
+                        (performance as any).webkitNow :
```

Comment:
It is still possible (although unlikely) that performance is defined and non of the methods we listed work, ie: now, mozNow, etc all undefined. Which means right at the end we should still have Date.now as a fall back there.